### PR TITLE
Wikis/user

### PR DESCRIPTION
**Pull request by AshakaE**
# Wikis created/edited per user

#### wikisCreated
image

#### wikisEdited
image

#### wiki author
image

## Notes or observations
_Query runs on Activity table, so it returns Activity[] not Wiki[]_

_An option to use wikiId result from Activity query above to query Wiki table for Wiki[]_

## Linked issues

closes https://github.com/EveripediaNetwork/issues/issues/404

### CODE REVIEW

File: src/App/user.resolver.ts [hunk 1]
```diff
@@ -37,6 +38,38 @@ class UserResolver {
...
   }
+
+  @ResolveField()
+  async wikisCreated(@Parent() user: IUser, @Args() args: PaginationArgs) {
+    const { id } = user
+    const repository = this.connection.getRepository(Activity)
+    return repository
+      .createQueryBuilder('activity')
+      .where(
+        'activity.datetime IN(SELECT MAX(activity.datetime) FROM activity GROUP BY activity.wikiId)',
```
Comment (kesar) [inline src/App/user.resolver.ts:49]: do u need this where and this group by for wikis created? can a wiki have two values CREATED?

Comment (AshakaE) [reply src/App/user.resolver.ts:49]: That will return only the wikiId column, I set it like that to get latest of the duplicated ids and return all columns

Comment (kesar) [reply src/App/user.resolver.ts:49]: there are not duplicated for status created. wikis can only be created once

Comment (AshakaE) [reply src/App/user.resolver.ts]: true, it should be for only Updated

**Comment by kesar** [comment]
i think its still missing wiki creator resolver

File: src/App/wiki.resolver.ts [hunk 2]
```diff
@@ -124,6 +126,18 @@ class WikiResolver {
...
+  @ResolveField()
+  async author(@Parent() wiki: IWiki) {
+    const { id } = wiki
+    const repository = this.connection.getRepository(Activity)
+    const res = await repository.query(`
+      SELECT "userId"
+      FROM "activity"
+      WHERE "wikiId" = '${id}' AND "type" = '0'
+    `,)
+    return { id: res[0].userId }
```
Comment (kesar) [inline src/App/wiki.resolver.ts:139]: does this return a User ?

File: src/Database/Entities/user.entity.ts [hunk 3]
```diff
@@ -18,6 +19,14 @@ class User implements IUser {
   @Field(() => [Wiki])
   @OneToMany('Wiki', 'user', { lazy: true })
   wikis!: IWiki[]
+
+  @Field(() => [Activity])
```
Comment (kesar) [inline src/Database/Entities/user.entity.ts:23]: are these entity fields required?

File: src/Database/Entities/wiki.entity.ts [hunk 4]
```diff
@@ -83,6 +84,10 @@ class Wiki {
   @ManyToOne('User', 'user', { lazy: true })
   user!: User
 
+  @Field(() => User)
```
Comment (kesar) [inline src/Database/Entities/wiki.entity.ts:87]: are these entity fields required?

File: src/Database/Entities/user.entity.ts (hunk 3 above)
Comment (AshakaE) [reply src/Database/Entities/user.entity.ts:23]: This one is needed, it throws an error without it

File: src/Database/Entities/wiki.entity.ts (hunk 4 above)
Comment (AshakaE) [reply src/Database/Entities/wiki.entity.ts:87]: This one seems fine without it, I'll reomve it

File: src/App/wiki.resolver.ts (hunk 2 above)
Comment (kesar) [reply src/App/wiki.resolver.ts:139]: I dont understand your response.

"yes, it returns the user as [{userId: '0x...id'}] from activty" <- [{userId: '0x...id'}] is not an User entity

File: src/Database/Entities/user.entity.ts (hunk 3 above)
Comment (kesar) [reply src/Database/Entities/user.entity.ts:23]: > This one is needed, it throws an error without it

Ok, but that's not responding my question. Does this generates new aux tables / columns on the DB?

Comment (AshakaE) [reply src/Database/Entities/user.entity.ts:23]: no aux tables generated

**Review by kesar — APPROVED** [review]

**Comment by kesar** [comment]
thanks! lets go!

**Comment by AshakaE** [comment]
> thanks! lets go!

😂

**Pull request merged**